discovery/syncer.go

```diff
@@ -854,13 +862,31 @@ func (g *GossipSyncer) replyChanRangeQuery(query *lnwire.QueryChannelRange) erro
 				"size=%v", g.cfg.peerPub[:], len(channelChunk))
 		}
 
+		// If we have any channels at all to return, then we need to
+		// update our pointers to the first and last blocks for each
+		// response.
+		if len(channelChunk) > 0 {
+			firstBlockHeight = channelChunk[0].BlockHeight
+			lastBlockHeight = channelChunk[len(channelChunk)-1].BlockHeight
+		}
+
+		// The number of blocks contained in this response (the total
+		// span) is the difference between the last channel ID and the
+		// first in the range. We add one as even if all channels
+		// returned are in the same block, we need to count that.
+		numBlocksInResp := lastBlockHeight - firstBlockHeight + 1
```

Comment:
I might be totally off here, but to me it seems that the result of `ChannelDB.FetchChannelRange` is not ordered as it is simply a filter over a cursor which iterates a bucket (`edgeIndex`). If that is true then `lastBlockHeight` might not be larger than `firstBlockHeight` and the above will overflow as both are `uint32`.

Reply:
IIRC the keys are sorted in ascending order in bolt, so this should be fine.

Reply:
Thanks @wpaulino, I missed that the Cursor iterates in order. (https://godoc.org/github.com/boltdb/bolt#Cursor)